internals/overlord/servstate/handlers.go

```diff
@@ -346,9 +346,16 @@ func (s *serviceData) startInternal() error {
 	s.cmd.Dir = s.config.WorkingDir
 
 	// Start as another user if specified in plan.
-	uid, gid, err := osutil.NormalizeUidGid(s.config.UserID, s.config.GroupID, s.config.User, s.config.Group)
-	if err != nil {
-		return err
+	w := s.manager.workloads[s.config.Workload]
+	uid, gid := w.UID, w.GID
+	if (uid == nil) != (gid == nil) {
+		panic("both uid and gid must be provided by workload")
```

Comment:
This panic will be triggered if either `uid` or `gid` is `nil` but not both. The panic text doesn't match.